src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -461,11 +461,11 @@ internal async Task<(IReadOnlyList<T> results, string continuationToken)> Execut
                     if (page.Count > 0)
                     {
                         results.AddRange(page);
-                        if (mustNotExceedMaxItemCount && results.Count > feedOptions.MaxItemCount)
+                        if (mustNotExceedMaxItemCount && page.Count > feedOptions.MaxItemCount)
```

Comment:
I had to read this a few times. Can we add a comment, i.e. the page returned more items than we expected...

Reply:
I honestly don't understand this code, and why we even want to throw things away and in what condition it should happen.

I added comment to what is going on right now with test.

Reply:
I am also not familiar with this code, but I wonder whether this PR #1439 can help put things in context

Reply:
So take these two examples:
1. Query the DB sequentially, we receive the max or less items then we specify as we scan through the partitions.
2. We issue the query in parallel, we get back results that may be equal to or less than the number items specified, but because it has been issued in parallel, there could be many pages more then we need

Reply:
Sorry, I don't understand your comment.